hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/filtering/FilteredMessageHandler.java

```diff
@@ -41,6 +47,8 @@ public void handle(FilterResult result, Message message, Subscription subscripti
             if (subscription.isTrackingEnabled()) {
                 trackers.get(subscription).logFiltered(toMessageMetadata(message, subscription), result.getFilterType().get());
             }
+
+            consumerRateLimiter.acquire();
```

Comment:
How does it work with heartbeat mode? If subscriber has some major failure and by design receives only 20% of messages on given topic then we are filtering 80% of messages very slowly and it would take huge amount of time for output rate to return to normal.